src/tests/ftest/launch.py

```diff
@@ -276,141 +315,156 @@ def get_available_interfaces(args):
                     else:
-                        active_interfaces[interface].update(nodeset)
+                        active_interfaces[interface].update(data.hosts)
                 except IndexError:
                     pass
 
         # From the active interface dictionary find all the interfaces that are common to all hosts
-        log("Active network interfaces detected:")
+        log.debug("Active network interfaces detected:")
         common_interfaces = []
         for interface, node_set in active_interfaces.items():
-            log("  - {0:<8} on {1} (Common={2})".format(interface, node_set, node_set == all_hosts))
+            log.debug("  - %-8s on %s (Common=%s)", interface, node_set, node_set == all_hosts)
             if node_set == all_hosts:
                 common_interfaces.append(interface)
 
         # Find the speed of each common active interface in order to be able to choose the fastest
         interface_speeds = {}
         for interface in common_interfaces:
-            speed = None
-            command = "cat {}".format(os.path.join(net_path, interface, "speed"))
-            task = get_remote_output(all_hosts, command)
-            if check_remote_output(task, command):
-                # Verify each host has the same interface speed
-                output_data = list(task.iter_buffers())
-                if len(output_data) > 1:
-                    log(
-                        "ERROR: Non-homogeneous interface speed detected for {} on {}.".format(
-                            interface, all_hosts))
-                else:
-                    for line in output_data[0][0]:
-                        try:
-                            interface_speeds[interface] = int(line.strip())
-                        except IOError as io_error:
-                            # KVM/Qemu/libvirt returns an EINVAL
-                            if io_error.errno == errno.EINVAL:
-                                interface_speeds[interface] = 1000
-                        except ValueError:
-                            # Any line not containing a speed (integer)
-                            pass
+            command = " ".join(["cat", os.path.join(net_path, interface, "speed")])
+            result = run_remote(log, all_hosts, command)
+            # Verify each host has the same interface speed
+            if result.passed and result.homogeneous:
+                for line in result.output[0].stdout:
+                    try:
+                        interface_speeds[interface] = int(line.strip())
+                    except IOError as io_error:
+                        # KVM/Qemu/libvirt returns an EINVAL
+                        if io_error.errno == errno.EINVAL:
+                            interface_speeds[interface] = 1000
+                    except ValueError:
+                        # Any line not containing a speed (integer)
+                        pass
```

Comment:
What if multiple lines contain an integer value? How do we know which is "really" the speed?

Reply:
There shouldn't be as the /sys/class/net/interface/speed is a system file.  We're really only iterating across multiple lines of output in case we get some thing like:
```
run_remote: Running on host-[26,181] with a 120 second timeout: cat /sys/class/net/eth0/speed
log_output:   host-[26,181] (rc=0):
log_output:     agent key RSA SHA256:4GhjwEd78jpLmBjGfWiG60 returned incorrect signature type
log_output:     1000
```